src/pyhf/infer/calculators.py

```diff
@@ -156,23 +161,30 @@ def teststatistic(self, poi_test):
         """
         tensorlib, _ = get_backend()
 
-        teststat_func = qmu_tilde if self.qtilde else qmu
+        if self.use_q0:
+            teststat_func = q0
+            asimov_mu = 1.0
+        else:
 
-        qmu_v = teststat_func(
-            poi_test, self.data, self.pdf, self.init_pars, self.par_bounds
-        )
-        sqrtqmu_v = tensorlib.sqrt(qmu_v)
+            def teststat_func(*args, **kwargs):
+                "make function signature without poi_test to be consistent with q0"
```

Comment:
The reason for making a function signature without `poi_test` is that `q0` doesn't take a `poi_test` argument since it is always 0. The alternative would be to change `q0` to take a `poi_test` argument, regardless if it only makes sense to be 0.